Antlr verification

BillWagner opened the pull request:
This PR replaces the existing validate grammar with a version that uses a private tool that runs as a global tool.

Once this is merged, the ANTLR grammar validation passes (with 1 warning)

BillWagner committed:
add NuPkg to validate grammar

Update the validate grammar workflow to use a private package to extract and update the .g4 files.

This package runs as a global dotnet tool

BillWagner committed:
fix one ANTLR error

Found this while testing the new workflow.

jskeet commented on .github/workflows/dependencies/ReplaceAndAdd.md:
```diff
@@ -0,0 +1,181 @@
+
+# Verification-Only Replacements & Additions
+
+This set of replacements and additions is the bare minimum required to allow the grammar
```
Rewrap?

jskeet commented on .github/workflows/dependencies/ReplaceAndAdd.md:
```diff
@@ -0,0 +1,181 @@
+This change resolves the one remaining MLR group by inlining some of the non-terminal
+alternatives in *primary_no_array_creation_expression*.
+
+Non-terminals that are inlined
```
Unwrap?

jskeet commented on .github/workflows/dependencies/ReplaceAndAdd.md:
```diff
@@ -0,0 +1,181 @@
+are commented out and the inlined body is indented.
+
+This change has not been made to the Standard itself as it makes *primary_no_array_creation_expression*
+“uglier” and would obfuscate somewhat the description in the Standard – both
```
Possibly remove the last part (from "both" onwards)?

BillWagner committed:
A bit of cleanup

- Effects of changes in tool:
  - More specific package name
  - More specific tool name
  - More specific version
- Fix warnings found in the grammar
- Fix one line in the script that was creating an unused file.
- Update .gitignore for temp files created by this tool

BillWagner committed:
one more "Identifier" casing issue

jskeet reviewed (approved):
Apologies, it looks like I failed to actually send this review...

BillWagner committed:
respond to feedback.
